### DISCUSSION TITLE
Extend Documentation: special tips for i.e. flask

### DISCUSSION THREAD

**Discussion by testor2897** [pyinstaller/pyinstaller#7581] [Features & Development]
Hello,

I am positively surprised how easy and smooth the PyInstaller works. You have really done a superb job.

If possible it would be nice if you could extend the documentation for some special applications.
As I had to figure out the best way for my flask (flaskwebgui) application I would like to suggest to give hints to the users which use these modules in combination with pyinstaller.

In my flaskwebgui case (Windows 10, Python 3.7.9) I had two extra challenges:
-	include 2 directories including subdirectories and files
-	make the current path usably within the packaged application

In order to make a correct package with PyInstaller I edited the spec file:
`
a = Analysis(
  ['app.py'],
  …
  datas=[('./static', 'static'), ('./templates', 'templates')],
  …
)
`
app.py is my main module. In the first part of datas statement I had to start with “./” to use direct subdirectories of app.py (is this windows specific?) - documented (https://pyinstaller.org/en/stable/spec-files.html) was "/". 

> P.S.: I just learned that starting without a slash is also working.

my flask/flaskwebgui framework in app.py is:  
```
import os
from flask import Flask, escape, render_template, request, session
# import webbrowser
from flaskwebgui import FlaskUI
...
app = Flask(__name__)
...
@app.route('/')
...

# change current directory to package root directory
os.chdir(app.root_path)

# flask version (old)
# webbrowser.open('http://localhost:8080')
# app.run(host='0.0.0.0', port=8080)
  
# flaskwebgui version (current)
FlaskUI(app=app, server="flask", port=8080).run()

```

In order to use the directories “static” and “templates” (subdirectories of app.py directory) as flask is expecting it I changed the current directory to package root directory.
`os.chdir(app.root_path)`

So I suggest to hint the users to the subdirectories example (maybe with differences between linux and windows?) and to the path topic.
Unfortunately I have no clue of the corresponding linux formulations.

Thank you very much for the great PyInstaller,

Sebastian

**Comment by testor2897** [comment]
What do you think about this suggestion (extension of documentation)?

**Reply by rokm** [reply]
A google search for PyInstaller + flask turns out several resources already (both instruction articles and code examples), so I don't see much benefit in extending the official documentation. Especially since someone would then have to maintain it to prevent it getting outdated and containing half truths and obsolete practices.

For specific projects/frameworks, there's also (community maintained) recipes pages on project's wiki: https://github.com/pyinstaller/pyinstaller/wiki/Recipes

**Comment by testor2897** [comment]
Great,
maybe a flask/flakswebgui recipe would be nice.

> P.S. I added a recipe for flask.
